plugins/config/src/subcommands/set/set.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	common "github.com/dokku/dokku/plugins/common"
+	"github.com/dokku/dokku/plugins/config/src/configenv"
+)
+
+// set the given entries from the specified environment
+func main() {
+	args := flag.NewFlagSet("config:set", flag.ExitOnError)
+	global := args.Bool("global", false, "--global: use the global environment")
+	encoded := args.Bool("encoded", false, "--encoded: interpret VALUEs as base64")
+	noRestart := args.Bool("no-restart", false, "--no-restart: no restart")
+	args.Parse(os.Args[2:])
+	shouldRestart := !*global && !*noRestart
+	var nextArg = 0
+	appName := args.Arg(0)
+	if appName == "" && !*global {
+		common.LogFail("Please specify an app or --global")
+	}
+
+	if *global {
+		appName = "--global"
+	} else {
+		nextArg = 1
+	}
+
+	env, err := configenv.NewFromTarget(appName)
+	if err != nil {
+		common.LogFail(err.Error())
+	}
+
+	var changed = false
+	entries := args.Args()[nextArg:]
+	for _, e := range entries {
+		//log
+		parts := strings.SplitN(e, "=", 2)
+		if len(parts) == 1 {
+			common.LogFail("Invalid env pair: " + e)
+		}
+		key, value := parts[0], parts[1]
+		if *encoded {
+			decoded, err := base64.StdEncoding.DecodeString(value)
+			if err != nil {
+				common.LogFail(fmt.Sprintf("%s for key '%s'", err.Error(), key))
+			}
+			value = string(decoded)
+		}
+		env.Set(key, value)
+		changed = true
+	}
+
+	if changed {
+		//log
+		env.Write()
+		args := append([]string{appName, "set"}, entries...)
+		common.PlugnTrigger("post-config-update", args...)
+	}
+
+	if shouldRestart && env.GetBoolDefault("DOKKU_APP_RESTORE", true) {
+		cmd := common.NewTokenizedShellCmd("dokku", "ps:restart", appName)
```

Comment:
I view plugin triggers as integration points; i.e. places where we want to allow something else to take over. I definitely see where this might be useful in the future but I'm not convinced now is the right time to do that. I'd rather not convolute the golang port effort with a shift in how core functionality interacts. I got in some trouble doing that on the python port. I'd be :+1: for a follow up PR that does that perhaps.